### PR TITLE
unit-tests as a Github action

### PR DESCRIPTION
**Pull request by kirienko**
I think this PR does the thing:
* triggering unit-tests;
* triggering on success
* adding a badge to the `README.md` page

But I feel like I'm missing something...

### CODE REVIEW

**Comment by RobertRosca** [comment]
Reason you had to use sudo for some of those commands was because you're missing the python setup step, also it would be nice if this runs the tests for multiple versions of python as well.

I'll make suggestions on the code so you can just merge them from here.

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
```
Comment (RobertRosca) [inline .github/workflows/unit-tests.yml:13]:
```suggestion
    strategy:
      fail-fast: false
      matrix:
        python-version: [3.7, 3.8]
```

Create matrix of different python versions

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install dependencies
```
Comment (RobertRosca) [inline .github/workflows/unit-tests.yml:18]:
```suggestion
      - uses: actions/checkout@v2

      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python-version }}

      - name: Install dependencies
```

Use the setup-python action to install python

File: .github/workflows/unit-tests.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        run: |
+          sudo python3 -m pip install --upgrade pip
+          pip install -U setuptools
+          sudo pip3 install nbconvert pelican pytest-tornasync
```
Comment (RobertRosca) [inline .github/workflows/unit-tests.yml:22]:
```suggestion
          python3 -m pip install --upgrade pip
          pip3 install -U setuptools
          pip3 install nbconvert pelican pytest-tornasync
```

With setup-python providing the python sudo shouldn't be required anymore

File: .github/workflows/unit-tests.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+      - name: Install oscovida
+        run: |
+          make dev-install
```
Comment (RobertRosca) [inline .github/workflows/unit-tests.yml:26]: Personally I dislike using makefiles for this since it can obfuscate errors and issues if they come up

Comment (RobertRosca) [reply .github/workflows/unit-tests.yml:26]: Also, for these tests, the install command should be the same one used by the users, so there's an actual reason for this in addition to my personal dislike of it :stuck_out_tongue:

**Review by RobertRosca — CHANGES_REQUESTED** [review]

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (kirienko) [reply .github/workflows/unit-tests.yml:20]: I didn't think there are any! Thanks!

File: .github/workflows/unit-tests.yml (hunk 4 above)
Comment (kirienko) [reply .github/workflows/unit-tests.yml:33]: Agree with you. My first thought was to get rig of that, but then I realised that in fact this `Makefile` is more visible to users that a Github action.

**Comment by RobertRosca** [comment]
Oh yes, also remove the old travis file to disable those tests since they're not needed anymore.

**Comment by kirienko** [comment]
Thanks for reviewing!

**Pull request merged**